hokusai/services/deployment.py

```diff
@@ -164,19 +165,31 @@ def update(self, tag, constraint, git_remote, timeout, update_config=False, file
 
       remote = git_remote or config.git_remote
       if remote:
-        print_green("Pushing Git deployment tags to %s..." % remote, newline_after=True)
-        try:
-          shout("git fetch %s --tags" % remote)
-          shout("git tag -f %s %s" % (self.context, tag), print_output=True)
-          shout("git tag -f %s %s" % (deployment_tag, tag), print_output=True)
-          shout("git push -f --no-verify %s refs/tags/%s" % (remote, self.context), print_output=True)
-          print_green("Updated Git tag %s -> %s" % (tag, self.context))
-          shout("git push -f --no-verify %s refs/tags/%s" % (remote, deployment_tag), print_output=True)
-          print_green("Updated Git tag %s -> %s" % (tag, deployment_tag), newline_after=True)
-        except CalledProcessError as e:
-          print_yellow("WARNING: Creating Git deployment tags failed due to the error: '%s'" % str(e), newline_before=True, newline_after=True)
-          print_yellow("The tag %s has been rolled out.  However, you should create the Git tags '%s' and '%s' manually, or re-run this deployment." % (tag, deployment_tag, self.context), newline_after=True)
-          post_deploy_success = False
+        # Update git tags. Try up to 3 times, at 3 second intervals. Failure does not fail deployment.
+        git_tag_sucess = False
+        attempts = 0
+        while ((not git_tag_sucess) and (attempts < 3)):
+          try:
+            attempts += 1
+            print_green("Creating Git deployment tags '%s', '%s', and push them to %s..." % (self.context, deployment_tag, remote))
+            print_green("Attempt# %s." %attempts)
+            shout("git fetch %s --tags" % remote)
+            shout("git tag -f %s %s" % (self.context, tag), print_output=True)
+            shout("git tag -f %s %s" % (deployment_tag, tag), print_output=True)
+            shout("git push -f --no-verify %s refs/tags/%s" % (remote, self.context), print_output=True)
+            print_green("Updated Git tag %s -> %s" % (tag, self.context))
+            shout("git push -f --no-verify %s refs/tags/%s" % (remote, deployment_tag), print_output=True)
+            print_green("Updated Git tag %s -> %s" % (tag, deployment_tag), newline_after=True)
+          except CalledProcessError as e:
+            print_yellow("WARNING: Creating Git deployment tags failed due to the error: '%s'" % str(e), newline_before=True, newline_after=True)
```

Comment:
👍 on logging the `output` attribute for the CalledProcessError raised.  I think we can drop this line that logs `str(e)`

Reply:
done.